### PR TITLE
Check if serverTags as configured in tierConfigs are valid

### PR DESCRIPTION
**Pull request by klsince**
Sanity check on serverTags as set in tierConfigs, to make sure they are valid

### CODE REVIEW

**Review by snleee — APPROVED** [review]
LGTM otherwise

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerStatelessTest.java [hunk 1]
```diff
@@ -540,6 +540,19 @@ public void testValidateTenantConfig() {
     // Valid tenant config without tagOverrideConfig
     _helixResourceManager.validateTableTenantConfig(offlineTableConfig);
 
+    // Invalid serverTag in tierConfigs
+    TierConfig tierConfig = new TierConfig("myTier", TierFactory.TIME_SEGMENT_SELECTOR_TYPE, "10d", null,
+        TierFactory.PINOT_SERVER_STORAGE_TYPE, "Unknown_OFFLINE", null, null);
```
Comment (walterddr) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerStatelessTest.java:545]: add a test for null? (looks like TierConfig json property server tag is nullable)

**Review by walterddr — APPROVED** [review]
lgtm. left one questions

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerStatelessTest.java (hunk 1 above)
Comment (klsince) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerStatelessTest.java:545]: good point. added.

**Pull request merged**